packages/demand-bloc-track-trip/src/TripBloc.ts

```diff
@@ -271,35 +271,37 @@ export class TripBloc {
    * Handle pagination in trips request
    */
   getNextUpcomingTrips() {
-    return this.getUpcomingTrips(this.tripsOffset.upcomingTripsOffset)
+    return this.getUpcomingTrips({ paginationOffset: this.tripsOffset.upcomingTripsOffset })
   }
 
   /**
    * The same method as getUpcomingTrips but for past trips list
    */
-  async getPastTrips(
+  async getPastTrips({
     paginationOffset = 0,
-    statuses: TripStatuses[] = [TripStatuses.COMPLETED],
-    paginationRowCount: number = this.options.paginationRowCount
-  ) {
+    statuses = [TripStatuses.CONFIRMED, TripStatuses.REQUESTED],
+    paginationRowCount = this.options.paginationRowCount,
+  }: TripsSearchParams) {
     return this.getTrips(this.pastTrips$, statuses, 'pastTripsOffset', paginationOffset, paginationRowCount)
   }
 
   /**
    * The same method as getNextUpcomingTrips but for past trips list
    */
   getNextPastTrips() {
-    return this.getPastTrips(this.tripsOffset.pastTripsOffset)
+    return this.getPastTrips({ paginationOffset: this.tripsOffset.pastTripsOffset })
   }
 
   /**
    * Cancels a trip by id
    * @param id trip id
    * @param params CancellationParams
    */
-  async cancel(id: string, params: CancellationParams) {
-    const response = await this.tripService.cancel(id, params)
+  async cancel(id: string, cancellationParams: CancellationParams, tripsSearchParams?: TripsSearchParams) {
+    const response = await this.tripService.cancel(id, cancellationParams)
 
-    if (response.status === 204) return this.getUpcomingTrips()
+    if (response.status === 204) {
+      return this.getUpcomingTrips(tripsSearchParams || {})
+    }
   }
```

Comment:
The method violates the single responsibility principle. I'd propose to discuss possible solution